### PR TITLE
buildLuarocksPackage: save luarocks config as derivation

### PR DESCRIPTION
**Pull request by teto**
while debugging luarocks packages, it's exhausting to have to build them, look at what random folder they've been built to finally look for their config.

With this you can run 
` nix build lua51Packages.plenary-nvim.configFile -f .` and infer what luarocks will want to do.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/interpreters/lua-5/build-luarocks-package.nix [hunk 1]
```diff
@@ -111,6 +112,12 @@ let
   # explicitly inherit this for it to be available as a shell variable in the
   # builder
   rocksSubdir = "${self.pname}-${self.version}-rocks";
+
+  configFile = writeTextFile {
+    name = pname + "-luarocks-config";
```
Comment (mrcjkb) [inline pkgs/development/interpreters/lua-5/build-luarocks-package.nix:117]: suggestion(probably doesn't matter):

```suggestion
    name = pname + "-luarocks-config.lua";
```

**Review by mrcjkb — APPROVED** [review]

File: pkgs/development/interpreters/lua-5/build-luarocks-package.nix (hunk 1 above)
Comment (teto) [reply pkgs/development/interpreters/lua-5/build-luarocks-package.nix]: good suggestion ty

**Comment by Artturin** [comment]
nixpkgs eval memory usage has gone up in the recent days, I found this pr by looking at [the search](https://github.com/NixOS/nixpkgs/pulls?q=is%3Apr+is%3Aclosed+base%3Amaster+NOT+in%3Atitle+Backport+) and this seems to be a possible cause as it adds 100K(estimation by @Kranzes) derivations?

**Comment by teto** [comment]
apologies if this caused issues. 
How does this cause 100K evals ?  maintainers/scripts/luarocks-packages.csv is 118 lines so we have ~ N ~ 100 lua packages per lua interpreter, of which we have ~10 . so 10N. 
This PR adds one derivation per lua package so it doubles the number of derivations so I would think this adds 1000 derivations (per platform) ? even if I missed sthg 100K seems like a lot ? or do I misunderstand how this works ?

**Comment by Kranzes** [comment]
I said that nixpkgs has 100k packages, not that this PR adds 100k packages. Was there a misunderstanding?

**Comment by Artturin** [comment]
Yeah

Screenshot_20231126_041708_Element

If this only doubles the derivation amount then it shouldn't cause such noticeable changes .


You could optimize by getting rid of the derivations and adding a function to lua which takes a luarock package and writes the config to a file, then the people who want to see the config file often can make a shell function which takes a luarock package and does `nix build --expr`

**Comment by Kranzes** [comment]
Oh I was replying to OP, sry.

**Pull request merged**